controller/node_controller.go

```diff
@@ -1137,7 +1138,24 @@ func (nc *NodeController) syncMultipathd(node *longhorn.Node, namespaces []lhtyp
 }
 
 func (nc *NodeController) checkKernelModulesLoaded(kubeNode *corev1.Node, node *longhorn.Node, namespaces []lhtypes.Namespace) {
-	notFoundModulesUsingkmod, err := checkModulesLoadedUsingkmod(kernelModules)
+	isV2DataEngine, err := nc.ds.GetSettingAsBool(types.SettingNameV2DataEngine)
+	if err != nil {
+		nc.logger.Errorf("Failed to fetch v2-data-engine setting: %v", err)
```

Comment:
Would be better to attach node information in the logger, and using structured logging.

```suggestion
	log := getLoggerForNode(nc.logger, node)
	isV2DataEngine, err := nc.ds.GetSettingAsBool(types.SettingNameV2DataEngine)
	if err != nil {
		log.WithError(err).Error("Failed to fetch v2-data-engine setting")
```